### DISCUSSION TITLE
Duda tiempo ejecución parte 1 con test hard

### DISCUSSION THREAD

**Discussion by melpony23** [IIC2133-PUC/2024-1#139] [Tarea 2] [answered]
Hola,

Tengo una duda respecto al tiempo de ejecución, ya que para el test hard1, mi output toma el siguiente tiempo en ejecutarse:

real    0m17.157s
user    0m16.156s
sys     0m1.000s

¿Está bien ese tiempo o debería ser más rápido? Usé el Rolling Hash (algoritmo de Rabin-Karp) y siento que ya he optimizado el tiempo lo máximo posible. Entiendo que el test debería ejecutarse en menos de 10 segundos, por eso no estoy segura.

¡Gracias!

**Comment by benjavicente** [comment]
Hola! Sube tu repositorio para revisar tu implementación.

**Reply by melpony23** [reply]
Okey! ya lo subí.

**Comment by benjavicente** [comment, marked as answer]
Corriendo tu código en mi computador, logra correr algo justo bajo los 10s, así que debería ser algo más rápido para que te cuenten esos tests bien. 

>  Usé el Rolling Hash (algoritmo de Rabin-Karp) y siento que ya he optimizado el tiempo lo máximo posible.

Te falta considerar aspectos específicos del problema de la tarea, como se relacionan a Rabin-Karp. Ojo que estas consideraciones son parte de lo que se pide explicar en el informe, como dice el enunciado 👀

**Reply by MaxAl100** [reply]
Todos podemos pedir este testeo? O cómo podemos tener un benchmark para ver cómo vamos en términos de velocidad?
Igualmente siento que he optimizado mi código, tomando:

> real    0m0.031s
> user    0m0.031s
> sys     0m0.000s

en mi dispositivo, pero tengo una torre que me armé, entonces es probable que sea mejor que el dispositivo promedio.

**Reply by benjavicente** [reply]
@MaxAl100 , si tu código corre en órdenes de magnitud de tiempo más rápido que los 10s, es improbable que corra en más de 10s en el sistema corrector.

**Reply by tomastrivino** [reply]
![97B30159-8699-4B4C-B324-79FC381621F5_4_5005_c](https://github.com/IIC2133-PUC/2024-1/assets/98418205/8002224e-7abf-4bc3-85bd-4d2675808c89)
